packages/anoncreds/src/utils/getRevocationRegistries.ts

```diff
@@ -164,38 +169,42 @@ export async function getRevocationRegistriesForProof(agentContext: AgentContext
       .resolve(AnonCredsRegistryService)
       .getRegistryForIdentifier(agentContext, revocationRegistryId)
 
-    // Fetch revocation registry definition if not already fetched
-    if (!revocationRegistries[revocationRegistryId]) {
-      const { revocationRegistryDefinition, resolutionMetadata } = await registry.getRevocationRegistryDefinition(
-        agentContext,
-        revocationRegistryId
-      )
-      if (!revocationRegistryDefinition) {
-        throw new AriesFrameworkError(
-          `Could not retrieve revocation registry definition for revocation registry ${revocationRegistryId}: ${resolutionMetadata.message}`
-        )
-      }
+    revocationRegistryPromises.push(
```

Comment:
same here, I find immediately executing function a bit hard to read often. cAn we extract to a separate method?